BootloaderCorePkg/Library/PciEnumerationLib/PciEnumerationLib.c

```diff
@@ -21,6 +21,22 @@
 
 UINT8   *mPoolPtr;
 
+//
+// Default PCI Resource Allocation Range
+//
+STATIC CONST PCI_RES_ALLOC_RANGE mDefaultResRange = {
+  .BusBase      = 0,

```

Comment:
do we need this default template ?
Can we always ask platform to provide the allocate table ?  It might give better control to platform.  EX:  MMIO32 limit is 4GB could be conflict with MMCFG base.    
We can even let our build tool to build default structure for the PCD.  So here we always assume the allocation table is available to make code a little bit simpler.

Reply:
Right, the resource limit could conflict with reserved memory and it varies on platforms. To simplify implementation, the resource limit has enough dummy size for now.
Do we need to enforce all platforms to provide all resource base/limit?
Regarding to default value, I initially did building the default structure in PCD, but the same c-type structure for resource allocation entries should be defined in the build script. Code is a little bit simpler, but duplicated definitions, so I put the default value in the code.

Reply:
OK, for now, let us keep this implementation like this.
For new platforms, we can ask them to always provide allocate table.  This is very useful for later phases too.  Even for UEFI payload, we might need to pass this info over.

Reply:
Got it. Let me revisit this to ask platforms to provide the table.